################################################################################
SHORT DESCRIPTION: 
################################################################################
Disabling interrupts early causes debug assertions.

################################################################################
LONG DESCRIPTION: 
################################################################################
This is only seen with debug builds but since it causes assertions it is
probably a bigger problem. It clearly says in map_pages_to_xen that it should
not be called with interrupts disabled. Moved disabling to just after that
call.

################################################################################
CHANGELOG 
################################################################################

################################################################################
REMOVAL 
################################################################################

################################################################################
UPSTREAM PLAN
################################################################################
Should be.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################

################################################################################
PATCHES 
################################################################################
--- a/xen/arch/x86/tboot.c
+++ b/xen/arch/x86/tboot.c
@@ -343,8 +343,6 @@ void tboot_shutdown(uint32_t shutdown_ty
 
     g_tboot_shared->shutdown_type = shutdown_type;
 
-    local_irq_disable();
-
     /* Create identity map for tboot shutdown code. */
     /* do before S3 integrity because mapping tboot may change xenheap */
     map_base = PFN_DOWN(g_tboot_shared->tboot_base);
@@ -359,6 +357,10 @@ void tboot_shutdown(uint32_t shutdown_ty
         return;
     }
 
+    /* Disable interrupts as early as possible but not before */
+    /* calling map_pages_to_xen */
+    local_irq_disable();
+
     /* if this is S3 then set regions to MAC */
     if ( shutdown_type == TB_SHUTDOWN_S3 )
     {
